### PR TITLE
Remove PageData.isCourseArchived and Logic.isCourseArchived #3747

### LINKED ISSUE #3747: Remove PageData.isCourseArchived and Logic.isCourseArchived
Doing this so that #3511 can be done more easily in future. After this is done, we should be able to just check for the instructor's isArchived to determine if a course is archived, removing the need for the method isCourseArchived, which checks both the course and instructor.

Things to be done here:
1. Update the migration script DataMigrationForIsArchived to add a preview mode, so we can check if the script had been ran before (Edit: this is done)
2. Add a default value for isArchived if the instructor entity is null.
3. Remove the PageData.isCourseArchive and related methods, and change the code calling them to just use the instructor isArchived attribute

At the end of the execution of the migration script, the instructor who are in an archived course should all have their isArchived attribute set. Instructor in old courses who are not archived will have isArchived as null, however at the instructorAttributes level, this will be set to false.

### PR DESCRIPTION
**Pull request by kanghj**
Fixes #3747. Part of the lead up to #3511. Will fix the existing tests' data in that issue.

### CODE REVIEW

File: src/test/resources/data/InstructorStudentListPageUiTest.json [hunk 1]
```diff
@@ -67,6 +63,7 @@
       "name": "Instructor1 Course1",
       "email": "instr@course1.tmt",
       "role":"Co-owner",
+      "isArchived":false,
       "isDisplayedToStudents":true,
       "displayedName":Instructor",
```
Comment (chowyb) [inline src/test/resources/data/InstructorStudentListPageUiTest.json]: Is there a missing double quote before `Instructor` here? Might be what's causing GitHub to display everything else in red.

Comment (wkurniawan07) [inline src/test/resources/data/InstructorStudentListPageUiTest.json]: Not sure how this affects stuffs, but it should be fine to fix it here.

File: src/main/java/teammates/ui/controller/PageData.java [hunk 2]
```diff
@@ -28,7 +28,6 @@
 import teammates.common.util.TimeHelper;
 import teammates.common.util.Url;
 import teammates.common.util.Utils;
-import teammates.logic.api.Logic;
```
Comment (wkurniawan07) [inline src/main/java/teammates/ui/controller/PageData.java]: :+1: finally this dubious dependency is removed.

**Comment by wkurniawan07** [comment]
@kanghj minor improvement requested.

**Comment by wkurniawan07** [comment]
Ok, looks good now :+1:

**Comment by wkurniawan07** [comment]
@kanghj correct me if I'm wrong, but doesn't this fix #3511 concurrently?

**Comment by kanghj** [comment]
Nope, this doesn't fix #3511 yet. The attribute hasn't been removed from CourseAttributes and Course yet, I just stopped using it. Removing it (and changing all the test data which still has a 'archive' attribute on Course) can be done separately just in case something went wrong in this PR.

**Pull request merged**